joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -519,4 +519,67 @@ trait AstForDeclarationsCreator(implicit withSchemaValidation: ValidationMode) {
     val methodRef = methodRefNode(lambdaExpression, code(lambdaExpression), fullName, lambdaReturnType)
     Ast(methodRef) :: Nil
   }
+
+  def astForAnonymousObjectCreationExpression(anonObjExpr: DotNetNodeInfo): Seq[Ast] = {
+    val typeDeclName     = nextAnonymousTypeName()
+    val typeDeclFullName = s"${scope.surroundingScopeFullName.getOrElse(Defines.Any)}.${typeDeclName}"
+
+    val _typeDeclNode = typeDeclNode(
+      anonObjExpr,
+      typeDeclName,
+      typeDeclFullName,
+      relativeFileName,
+      code(anonObjExpr),
+      astParentType = NodeTypes.METHOD,
+      astParentFullName = scope.surroundingScopeFullName.getOrElse(Defines.Any)
+    )
+
+    scope.pushNewScope(TypeScope(typeDeclFullName))
+
+    val memberAsts = anonObjExpr
+      .json(ParserKeys.Initializers)
+      .arr
+      .map(createDotNetNodeInfo)
+      .flatMap(astForAnonymousObjectMemberDeclarator)
+      .toSeq
+
+    scope.popScope()
+    Ast.storeInDiffGraph(Ast(_typeDeclNode).withChildren(memberAsts), diffGraph)
+
+    val _typeRefNode = typeRefNode(anonObjExpr, code(anonObjExpr), typeDeclFullName)
+    Ast(_typeRefNode) :: Nil
+  }
+
+  def astForAnonymousObjectMemberDeclarator(memberDeclarator: DotNetNodeInfo): Seq[Ast] = {
```

Comment:
If this always only returns one `Ast`, then it can simply return one, e.g.
```suggestion
  private def astForAnonymousObjectMemberDeclarator(memberDeclarator: DotNetNodeInfo): Ast = {
```